### PR TITLE
osd: zipkin tracing

### PR DESCRIPTION
**Pull request by liewegas**
These are just the librados and osd patches.  The librbd pieces still need to be rebased (and that will likely be a bit painful).

### CODE REVIEW

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-04-05_18:31:27-rados-wip-sage-testing---basic-smithi/

**Comment by cbodley** [comment]
@liewegas i found this cmake commit https://github.com/cbodley/ceph/commit/5e5ef94d855d798def3b6b54f439ed1af9582a94 that somehow got dropped from the previous pr

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
blkin needs to link against hte statically built boost... see https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=trusty,DIST=trusty,MACHINE_SIZE=huge/2448//consoleFull

**Comment by liewegas** [comment]
https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos7,DIST=centos7,MACHINE_SIZE=huge/2516//consoleFull

<pre>
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.5", minimum required is "2") 
-- Found PythonLibs: /usr/lib64/libpython2.7.so (found suitable version "2.7.5", minimum required is "2") 
-- Found Python3Interp: /usr/bin/python3 (found suitable version "3.4.3", minimum required is "3") 
-- Found Python3Libs: /usr/lib64/libpython3.4m.so (found suitable version "3.4.3", minimum required is "3") 
CMake Error at src/blkin/blkin-lib/CMakeLists.txt:9 (target_compile_options):
  Unknown CMake command "target_compile_options".


-- Configuring incomplete, errors occurred!
error: Bad exit status from /var/tmp/rpm-tmp.Ttf5Cq (%build)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.Ttf5Cq (%build)
</pre>

**Comment by liewegas** [comment]
on centos only...

**Comment by cbodley** [comment]
@liewegas https://github.com/ceph/blkin/pull/3 should fix that `Unknown CMake command "target_compile_options".` error

File: CMakeLists.txt [hunk 1]
```diff
@@ -344,8 +344,8 @@ endif(WITH_DPDK)
 
 option(WITH_BLKIN "Use blkin to emit LTTng tracepoints for Zipkin" OFF)
 if(WITH_BLKIN)
-  find_package(blkin REQUIRED)
-  include_directories(${BLKIN_INCLUDE_DIR})
+  set(BLKIN_LIBRARIES blkin lttng-ust)
+  include_directories(src/blkin/blkin-lib)
```
Comment (tchaikov) [inline CMakeLists.txt]: can we add blkin as an external project, so its targets won't be populated into ceph's cmake targets, and "make install" won't include the blkin?

**Comment by alimaredia** [comment]
@liewegas trying to resolve this build issue (https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos7,DIST=centos7,MACHINE_SIZE=huge/2544//consoleFull) here: https://github.com/ceph/blkin/pull/4 hopefully we can resolve it on monday when Casey's back

**Comment by yuriw** [comment]
* branch            wip-zipkin-partial -> FETCH_HEAD
Auto-merging src/osdc/Objecter.h
Auto-merging src/osd/ReplicatedBackend.cc
Auto-merging src/osd/PrimaryLogPG.cc
Auto-merging src/osd/PG.h
Auto-merging src/osd/PG.cc
Auto-merging src/osd/OSD.h
Auto-merging src/osd/OSD.cc
Auto-merging src/osd/ECBackend.cc
Auto-merging src/os/filestore/FileStore.cc
Auto-merging src/common/config_opts.h
Auto-merging src/CMakeLists.txt
Auto-merging .gitmodules
CONFLICT (content): Merge conflict in .gitmodules
Automatic merge failed; fix conflicts and then commit the result.

@liewegas fyi

**Comment by liewegas** [comment]
rebased

**Comment by liewegas** [comment]
The following tests FAILED:
          1 - testc (Not Run)
          2 - testpp (Not Run)
          3 - testppp (Not Run)

**Comment by alimaredia** [comment]
@liewegas this PR should fix those tests not being run: https://github.com/ceph/blkin/pull/6

**Comment by liewegas** [comment]
I'm able to make testpp fail or hang with 'ctest -R testp -V --repeat-until-fail 100 -j 10' (and testpp failed the first time around with a ctest -j12).

**Comment by liewegas** [comment]
rebased

**Comment by liewegas** [comment]
retest this please

**Comment by yuriw** [comment]
removed from test batch per @liewegas  lots of "SELinux denials"
http://pulpito.ceph.com/yuriw-2017-04-27_19:42:46-rados-wip-yuri-testing_2017_4_28-distro-basic-smithi/

**Comment by liewegas** [comment]
<pre>SELinux denials found on ubuntu@smithi032.front.sepia.ceph.com: ['type=AVC msg=audit(1493322598.789:167147): avc: denied { open } for pid=824540 comm="ceph-mon" path="/dev/shm/lttng-ust-wait-5" dev="tmpfs" ino=2417635 scontext=system_u:system_r:ceph_t:s0 tcontext=system_u:object_r:initrc_state_t:s0 tclass=file', 'type=AVC msg=audit(1493322619.318:167226): avc: denied { read } for pid=824756 comm="ceph-mon" name="lttng-ust-wait-5" dev="tmpfs" ino=2417635 scontext=system_u:system_r:ceph_t:s0 tcontext=system_u:object_r:initrc_state_t:s0 tclass=file', 'type=AVC msg=audit(1493322588.673:167112): avc: denied { read } for pid=824382 comm="ceph-mon" name="lttng-ust-wait-5" dev="tmpfs" ino=2417635 scontext=system_u:system_r:ceph_t:s0 tcontext=system_u:object_r:initrc_state_t:s0 tclass=file', 'type=AVC msg=audit(1493322619.318:167226): avc: denied { open } for pid=824756 comm="ceph-mon" path="/dev/shm/lttng-ust-wait-5" dev="tmpfs" ino=2417635 scontext=system_u:system_r:ceph_t:s0 tcontext=system_u:object_r:initrc_state_t:s0 tclass=file', 'type=AVC msg=audit(1493322588.673:167112): avc: denied { open } for pid=824382 comm="ceph-mon" path="/dev/shm/lttng-ust-wait-5" dev="tmpfs" ino=2417635 scontext=system_u:system_r:ceph_t:s0 tcontext=system_u:object_r:initrc_state_t:s0 tclass=file', 'type=AVC msg=audit(1493322598.789:167147): avc: denied { read } for pid=824540 comm="ceph-mon" name="lttng-ust-wait-5" dev="tmpfs" ino=2417635 scontext=system_u:system_r:ceph_t:s0 tcontext=system_u:object_r:initrc_state_t:s0 tclass=file']

</pre>
and more at http://pulpito.ceph.com/yuriw-2017-04-27_19:42:46-rados-wip-yuri-testing_2017_4_28-distro-basic-smithi/

@b-ranto mind taking a look?

**Comment by b-ranto** [comment]
This looks like the PR is enabling lttng tracepoints by default. We are building two versions of the libraries by default. A default library without the tracepoints and an lttng enabled one with a _tp suffix. The lttng enabled one does show these denials.

**Comment by liewegas** [comment]
@alimaredia @cbodley how should we proceed?

**Comment by cbodley** [comment]
@liewegas i was under the impression that we were already using the existing libraries under `ceph/src/tracing/`, since `WITH_LTTNG=ON` is already the default. have those not been tested in teuthology?

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-05-05_19:23:54-rados-wip-sage-testing---basic-smithi/

**Comment by wjwithagen** [comment]
@liewegas 
@tchaikov 

I can not fully fathom the discussion above, but ....
 
This introduces again the `target_link_libraries(common ...)` 
And with the libreplacements that are already in us for things like resolv.
(Which FreeBSD does not have) 

I'm assuming this addition is purposely, even though Kefu tried that to get `ceph-common` going?

If this is not an oversight and this needs to stay, I'll submit a PR that fixes certain of these references.
Otherwise we should remove it, since Cmake does not have a problem with it being commented out.

**Pull request merged**